tools/GSDumpGUI/Forms/frmMain.cs

```diff
@@ -107,15 +116,61 @@ public GSDumpGUI()
 
             _availableGsDumps.OnIndexUpdatedEvent += UpdatePreviewImage;
 
-            txtGSDXDirectory.DataBindings.Add(nameof(TextBox.Text), Settings, nameof(Settings.GSDXDir));
-            txtDumpsDirectory.DataBindings.Add(nameof(TextBox.Text), Settings, nameof(Settings.DumpDir));
+            this.Text += " " + (IntPtr.Size * 8).ToString() + "bits";
+
+            if (String.IsNullOrEmpty(Settings.GSDXDir) || !Directory.Exists(Settings.GSDXDir))
+                Settings.GSDXDir = AppDomain.CurrentDomain.BaseDirectory;
+
+            if (String.IsNullOrEmpty(Settings.DumpDir) || !Directory.Exists(Settings.DumpDir))
+                Settings.DumpDir = AppDomain.CurrentDomain.BaseDirectory;
+
+            txtGSDXDirectory.Text = Settings.GSDXDir;
+            txtDumpsDirectory.Text = Settings.DumpDir;
 
             BindListControl(lstDumps, _availableGsDumps, g => g.Files, f => f.DisplayText, g => g.SelectedFileIndex);
             BindListControl(lstGSDX, _availableGsDlls, g => g.Files, f => f.DisplayText, g => g.SelectedFileIndex);
 
             Processes = new List<Process>();
 
             NoImage = CreateDefaultImage();
+
+            _dllWatcher = new List<FileSystemWatcher>();
+            _dumpWatcher = new List<FileSystemWatcher>();
+            _watcherEvents = new ConcurrentQueue<int>();
+
+            Thread watcherThread = new Thread(ChangesWatchdogThread);
+            watcherThread.Start();
+        }
+
+        private void ChangesWatchdogThread()
+        {
+            while (Program.isProgRunning)
+            {
+                bool dllReload = false;
+                bool dumpReload = false;
+
+                int evt;
+                while (_watcherEvents.TryDequeue(out evt))
+                {
+                    if (evt == 1) dllReload = true;
```

Comment:
I’d use an enum for 1 and 2 to improve readability/maintainability

Reply:
Done!